framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -330,6 +340,25 @@ MeshDiagnosticsGenerator::checkWaterTightSidesets(const std::unique_ptr<MeshBase
             num_faces_without_sideset++;
           }
         }
+        else if (!_watertight_boundaries.empty() && intersections.empty())
```

Comment:
can you consolidate all of this code by doing:
if (...... OR (!_watertight_boundaries.empty() && intersections.empty()))

Reply:
I separated the two because I figured that was the best way to have different error messages depending on whether it just has no sideset, or was assigned to the wrong one? If we want them combined I could make the error message more generic?